python/GafferUI/RampPlugValueWidget.py

```diff
@@ -78,35 +78,34 @@ def __init__( self, plug, **kw ) :
 			self.__lastPositionChangedReason = None
 			self.__positionsMergeGroupId = 0
 
-			with GafferUI.ListContainer(
-				orientation = GafferUI.ListContainer.Orientation.Horizontal if isinstance( plug.pointYPlug( 0 ), Gaffer.FloatPlug ) else GafferUI.ListContainer.Orientation.Vertical,
-				spacing = 4
-			) :
-
-				self.__positionLabel = GafferUI.LabelPlugValueWidget(
-					plug.pointXPlug( 0 ),
-					parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-				)
-				self.__positionField = GafferUI.NumericPlugValueWidget(
-					plug.pointXPlug( 0 ),
-					parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-				)
-
-				self.__valueLabel = GafferUI.LabelPlugValueWidget(
-					plug.pointYPlug( 0 ),
-					parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-				)
-				if isinstance( plug.pointYPlug( 0 ), Gaffer.FloatPlug ):
-					self.__valueField = GafferUI.NumericPlugValueWidget(
-						plug.pointYPlug( 0 ),
-						parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-					)
-				else:
-					self.__valueField = GafferUI.ColorPlugValueWidget(
-						plug.pointYPlug( 0 ),
-						parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-					)
-					self.__valueField.setColorChooserVisible( True )
+			self.__positionLabel = GafferUI.LabelPlugValueWidget( plug.pointXPlug( 0 ) )
+			self.__positionField = GafferUI.NumericPlugValueWidget( plug.pointXPlug( 0 ) )
```

Comment:
Because we're inside a `with column` block here, these widgets are going to first be parented to that column before being pulled out and parented elsewhere by the `addChild()` calls below. I find the structured (`with`) style of UI building much more readable than the manual `addChild()` method, because you can see the structure of the UI in the structure of the code. So I'd suggest that the lesser evil would be to use `with` when constructing the containers below, even if it means two lines that create the position widgets.

Reply:
It's definitely possible to give a fixed width to the labels - the trick is to do what PlugWidget does, and set the width on the label inside the widget : `self.__positionLabel.label()._qtWidget().setFixedWidth( 400 )`. I think it would be worth doing this to get the green dots to align.

Reply:
Oh nice, thanks for that tip on `label()._qtWidget()`, it's helpful. I added that in https://github.com/GafferHQ/gaffer/pull/5737/commits/5cdd19a9fb0714f0d8d7db9b9a273f68e13ba497

Reply:
> I find the structured (`with`) style of UI building much more readable than the manual `addChild()` method

I agree, and definitely shouldn't be added and then removed from the parent container. I'll make that change, pending what we ultimately go with for the container choice discussed below.